typescript/schedules.md

```diff
@@ -0,0 +1,1157 @@
+ */
+export interface CalendarSpecDescription {
+  /**
+   * Valid values: 0–59
+   *
+   * @default `[{ start: 0 }]`
+   */
+  second?: NumberSpecDescription;
+
+  /**
+   * Valid values: 0–59
+   *
+   * @default `[{ start: 0 }]`
+   */
+  minute?: NumberSpecDescription;
+
+  /**
+   * Valid values: 0–59
+   *
+   * @default `[{ start: 0 }]`
+   */
+  hour?: NumberSpecDescription;
+
+  /**
+   * Valid values: 1–31
+   *
+   * @default `[{ start: 1, end: 31 }]`
+   */
+  dayOfMonth?: NumberSpecDescription;
+
+  /**
+   * @default `[{ start: 'JANUARY' , end: 'DECEMBER' }]`
+   */
+  month?: MonthSpecDescription;
+
+  /**
+   * Use full years, like `2030`
+   *
+   * @default `undefined` (meaning any year)
+   */
+  year?: NumberSpecDescription;
+
+  /**
+   * @default `[{ start: 'SUNDAY', end: 'SATURDAY' }]`
+   */
+  dayOfWeek?: DaySpecDescription;
+}
+
+/**
+ * IntervalSpec matches times that can be expressed as:
+ *
+ * `Epoch + (n * every) + offset`
+ *
+ * where `n` is all integers ≥ 0.
+ *
+ * For example, an `every` of 1 hour with `offset` of zero would match every hour, on the hour. The same `every` but an `offset`
+ * of 19 minutes would match every `xx:19:00`. An `every` of 28 days with `offset` zero would match `2022-02-17T00:00:00Z`
+ * (among other times). The same `every` with `offset` of 3 days, 5 hours, and 23 minutes would match `2022-02-20T05:23:00Z`
+ * instead.
+ */
+export interface IntervalSpec {
+  /**
+   * Value is rounded to the nearest second.
+   *
+   * @format number of milliseconds or {@link https://www.npmjs.com/package/ms | ms-formatted string}
+   */
+  every: number | string;
+
+  /**
+   * Value is rounded to the nearest second.
+   *
+   * @default 0
+   * @format number of milliseconds or {@link https://www.npmjs.com/package/ms | ms-formatted string}
+   */
+  offset?: number | string;
+}
+
+/**
+ * The version of {@link IntervalSpec} that you get back from {@link ScheduleHandle.describe}
+ */
+export interface IntervalSpecDescription {
+  /**
+   * Value is rounded to the nearest second.
+   *
+   * @format number of milliseconds
+   */
+  every: number;
+
+  /**
+   * Value is rounded to the nearest second.
+   *
+   * @default 0
+   * @format number of milliseconds
+   */
+  offset?: number;
+}
+
+/**
+ * A complete description of a set of absolute times (possibly infinite) that an Action should occur at.
+ * The times are the union of `calendars`, `intervals`, and `cronExpressions`, minus the `skip` times. These times
+ * never change, except that the definition of a time zone can change over time (most commonly, when daylight saving
+ * time policy changes for an area). To create a totally self-contained `ScheduleSpec`, use UTC.
+ */
+export interface ScheduleSpec {
+  /** Calendar-based specifications of times. */
+  calendars?: CalendarSpec[];
+
+  /** Interval-based specifications of times. */
+  intervals?: IntervalSpec[];
+
+  /**
+   * [Cron expressions](https://crontab.guru/)
```

Comment:
Add a note that this is provided for migration and that new applications should prefer `calendars` for readability and maintainability?

Reply:
👍